internal/template/column.go

```diff
@@ -170,9 +171,17 @@ func (c Column) View(ix index.Int) View {
 	return View{data: c.data, index: ix}
 }
 
-func (c Column) DataType() string {
-	var x dataType
-	return fmt.Sprintf("%v", reflect.TypeOf(x))
+func (c Column) DataType() types.DataType {
```

Comment:
It would probably be better to move this into a file that is not generated (eg. `column.go` in each types package). That will make the code a bit easier to follow I think. You would just have to return the data type the same way it is done for string and enum.

Reply:
Thanks, this approach makes a lot more sense.

Reply:
Sorry, spoke too quickly..

AFAICT it is actually not possible to move the `DataType()` methods of genny generated types into individual column packages because the `column.Column` type defined in `internal/template` package must also implement the `column.Column` interface from `internal/column` :roll_eyes: .

In some ways this makes me feel it would be better to forego genny all together and introduce a bit more duplication and/or extend the `qfgenerate` tool.

Thoughts?

Reply:
The interface compatibility issue should be solvable by adding an implementation of `DataType()` into `placeholders.go`. Similar issues have popped up before.

In general I agree with you that the line between what to generate and what to copy and paste is somewhat fuzzy and that it is a bit confusing to have half of the methods on the column generated and half of them written individually, by hand, for each type.
We might consider removing the genny-templates at some point in the future in favor of duplicating the code. I don't quite feel we're there yet though. There may be other, even better, ways of handling this duplication.

Reply:
Thanks for the feedback. It took me a bit of time to understand more deeply how genny was being used. I followed your suggestion of using `placeholders.go` and that has simplified the code.